tests/jest/unit/assets/dev/js/editor/regions/navigator/components/header.test.js

```diff
@@ -0,0 +1,82 @@
+import { render, fireEvent } from '@testing-library/react';
+import { Header } from 'elementor-regions/navigator/components';
+
+const mockSetElementFolding = jest.fn(),
+	mockStore = {
+		'navigator/folding': {},
+	};
+
+jest.mock( 'elementor-assets-js/editor/regions/navigator/hooks', () => ( {
+	useElementFolding: () => [ mockStore[ 'navigator/folding' ], mockSetElementFolding ],
+} ) );
+
+describe( '<Header />', () => {
+	beforeEach( () => {
+		jest.clearAllMocks();
+	} );
+
+	it( 'Should render navigator title', () => {
+		const component = render( <Header /> );
+
+		expect(
+			component.queryByText( 'Navigator' )
+		).toBeInTheDocument();
+	} );
+
+	it( 'Should close navigator on close click', () => {
+		const handleClose = jest.fn(),
+			component = render( <Header onClose={ handleClose } /> );
+
+		fireEvent.click( component.getByRole( 'close' ) );
+
+		expect( handleClose ).toBeCalledTimes( 1 );
+	} );
+
+	it( 'Should activate all items folding on toggle-all click when some elements are not selected', () => {
+		mockStore[ 'navigator/folding' ][ 'test-element-1' ] = true;
+		mockStore[ 'navigator/folding' ][ 'test-element-2' ] = false;
+
+		const component = render( <Header /> );
+
+		fireEvent.click( component.getByRole( 'toggle-all' ) );
+
+		expect( mockSetElementFolding ).toBeCalledTimes( 1 );
+		expect( mockSetElementFolding ).toBeCalledWith( true );
+
+		// Cleanup
+		delete mockStore[ 'navigator/folding' ][ 'test-element-1' ];
+		delete mockStore[ 'navigator/folding' ][ 'test-element-2' ];
+	} );
+
+	it( 'Should activate all items folding on toggle-all click when all elements are not selected', () => {
+		mockStore[ 'navigator/folding' ][ 'test-element-1' ] = false;
+		mockStore[ 'navigator/folding' ][ 'test-element-2' ] = false;
+
+		const component = render( <Header /> );
+
+		fireEvent.click( component.getByRole( 'toggle-all' ) );
+
+		expect( mockSetElementFolding ).toBeCalledTimes( 1 );
```

Comment:
Since it tests the `header` component, which has the ability to toggle all elements selection, the `setElementFolding` function should be called only once - for updating all elements recursively. I'll change the name of the function to `seElementsFolding` so it will be clearer.